core/client/client.go

```diff
@@ -329,12 +329,12 @@ func (c *Client) waitOnceForUserUpdate(ctx context.Context, m *entry.Mutation) (
 		return nil, fmt.Errorf("nil mutation")
 	}
 	// Wait for STH to change.
-	if err := c.WaitForSTHUpdate(ctx, int64(c.trusted.TreeSize)+1); err != nil {
+	if err := c.WaitForSTHUpdate(ctx, m.PrevRevision()+1); err != nil {
```

Comment:
Consider creating a method Mutation.ExpectedRevision() or MinApplicableRevision(), which returns PrevRevision()+1.

I'm not completely sold on the elegance of this, but "PrevRevision + 1" does map to some logical concept which is a property of the mutation.

Reply:
Thanks -- that's a much better API.